File: 백준/Silver/1991. 트리 순회/트리 순회.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    static int n;
    static int child[][] = new int[27][3];

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(bf.readLine());

        char a, b, c;

        for(int i=1; i<=n; i++){
            String tmp = bf.readLine();
            a = tmp.charAt(0);
            b = tmp.charAt(2);
            c = tmp.charAt(4);
            if(b != '.') child[a-'A'+1][0] = b-'A'+1;
            if(c != '.') child[a-'A'+1][1] = c-'A'+1;
        }

        preorder(1);
        System.out.print("\n");
        inorder(1);
        System.out.print("\n");
        postorder(1);

    }

    private static void postorder(int i) {
        if(i == 0) return;
        postorder(child[i][0]);
        postorder(child[i][1]);
        System.out.print((char)('A'+i-1));
    }

    private static void inorder(int i) {
        if(i == 0) return;
        inorder(child[i][0]);
        System.out.print((char)('A'+i-1));
        inorder(child[i][1]);
    }

    private static void preorder(int i) {
        if(i == 0) return;
        System.out.print((char)('A'+i-1));
        preorder(child[i][0]);
        preorder(child[i][1]);
    }
}

File: 백준/Silver/1991. 트리 순회/트리 순회.cc
#include <iostream>
#include <vector>
#include <string>

using namespace std;
int num;
int arr[27][3];

void inorder(int n) {
	if (n == 0) return;

	inorder(arr[n][0]);
	char a = n + 'A' - 1;
	cout << a;
	inorder(arr[n][1]);
}

void preorder(int n) {
	if (n == 0) return;

	char a = n + 'A' - 1;
	cout << a;
	preorder(arr[n][0]);
	preorder(arr[n][1]);
}


void postorder(int n) {
	if (n == 0) return;

	postorder(arr[n][0]);
	postorder(arr[n][1]);
	char a = n + 'A' - 1;
	cout << a;
}

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);

	cin >> num;

	for (int i = 1; i <= num; i++) {
		char a, b, c;
		cin >> a >> b >> c;

		if (b != '.')arr[a - 'A' + 1][0] = b - 'A' + 1;
		if (c != '.')arr[a - 'A' + 1][1] = c - 'A' + 1;
	}

	preorder(1);
	cout << "\n";
	inorder(1);
	cout << "\n";
	postorder(1);

	return 0;
}